include/networkit/graph/GraphTools.hpp

```diff
@@ -418,6 +419,59 @@ void sortEdgesByWeight(Graph &G, bool decreasing = false);
  */
 std::vector<node> topologicalSort(const Graph &G);
 
+/**
+ * Given a  grah G, randomizes the weights in the graph. The weights are distributed in
+ * the range [0, 1] by default, unless a distribution is provided. This occurs in-place
+ * if the graph is already weighted, but creates a weighted copy before randomizing weights
+ * if it wasn't.
+ * @param   G           Directed input graph
+ * @param   distr       Random distribution
+ */
+template <class Distribution = std::uniform_real_distribution<edgeweight>>
+void randomizeWeights(Graph &G, Distribution distr = std::uniform_real_distribution<edgeweight>{
+                                    0, std::nexttoward(1.0, 2.0)}) {
+    if (!G.isWeighted())
+        G = toWeighted(G);
+#pragma omp parallel
+    {
+        std::mt19937 gen;
+        auto baseSeed = Aux::Random::integer();
+
+#pragma omp for schedule(dynamic)
+        for (omp_index u = 0; u < G.upperNodeIdBound(); u++) {
+            gen.seed(std::hash<node>()(u) + 0x9e3779b9 + (baseSeed << 6) + (baseSeed >> 2));
```

Comment:
Leave a comment what is done here (and why).

Reply:
Thanks for inserting comments - however the remark was about what the ingredients in the seed actual do. I think the combination of magic number and bit shift is taken from boost::combine_hash?

Also the `baseSeed` makes the result not deterministic if the number of threads is bigger >1, right?

Reply:
> Also the baseSeed makes the result not deterministic if the number of threads is bigger >1, right?

I tested and the results are deterministic